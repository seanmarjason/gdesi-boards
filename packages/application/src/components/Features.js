import * as React from 'react';
import PropTypes from 'prop-types';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Card from '@mui/material/Card';
import Chip from '@mui/material/Chip';
import Container from '@mui/material/Container';
import Typography from '@mui/material/Typography';
import { styled } from '@mui/material/styles';

import DevicesRoundedIcon from '@mui/icons-material/DevicesRounded';
import Dashboard from '@mui/icons-material/Dashboard';
import HowToRegIcon from '@mui/icons-material/HowToReg';
import SmartToyIcon from '@mui/icons-material/SmartToy';

const items = [
  {
    icon: <Dashboard />,
    title: 'Team Task Boards',
    description:
      'Collaborate with your team asynchronously, continuously capturing the latest updates on your work in a central location.',
    imageLight: `url("${process.env.TEMPLATE_IMAGE_URL || 'https://gdesi-boards.vercel.app'}/Dashboard.png")`,
    imageDark: `url("${process.env.TEMPLATE_IMAGE_URL || 'https://gdesi-boards.vercel.app'}/Dashboard.png")`,
  },
  {
    icon: <HowToRegIcon />,
    title: 'Check Ins',
    description:
      'Preiodically check in with your team manager to showcase your progress, and highlight your achievements or where you might need support.',
    imageLight: `url("${process.env.TEMPLATE_IMAGE_URL || 'https://gdesi-boards.vercel.app'}/Check-in.png")`,
    imageDark: `url("${process.env.TEMPLATE_IMAGE_URL || 'https://gdesi-boards.vercel.app'}/Check-in.png")`,  },
  {
    icon: <SmartToyIcon />,
    title: 'Two-way communication with OpenAI',
    description:
      'Harness the power of OpenAI to simulate two-way comnunication, ensuring you give the right amount of information to your team members when they need it.',
    imageLight: `url("${process.env.TEMPLATE_IMAGE_URL || 'https://gdesi-boards.vercel.app'}/OpenAI.png")`,
    imageDark: `url("${process.env.TEMPLATE_IMAGE_URL || 'https://gdesi-boards.vercel.app'}/OpenAI.png")`,  },
];

export default function Features() {
  const [selectedItemIndex, setSelectedItemIndex] = React.useState(0);

  const handleItemClick = (index) => {
    setSelectedItemIndex(index);
  };

  const selectedFeature = items[selectedItemIndex];

  return (
    <Container id="features" sx={{ py: { xs: 8, sm: 16 } }}>
      <Box sx={{ width: { sm: '100%', md: '60%' } }}>
        <Typography
          component="h2"
          variant="h4"
          gutterBottom
          sx={{ color: 'text.primary' }}
        >
          Product features
        </Typography>
        <Typography
          variant="body1"
          sx={{ color: 'text.secondary', mb: { xs: 2, sm: 4 } }}
        >
          Experience the latest features to improve how you work remotely.
        </Typography>
      </Box>
      <Box
        sx={{
          display: 'flex',
          flexDirection: { xs: 'column', md: 'row-reverse' },
          gap: 2,
        }}
      >
        <div>
          <Box
            sx={{
              display: { xs: 'none', sm: 'flex' },
              flexDirection: 'column',
              gap: 2,
              height: '100%',
            }}
          >
            {items.map(({ icon, title, description }, index) => (
              <Box
                key={index}
                component={Button}
                onClick={() => handleItemClick(index)}
                sx={[
                  (theme) => ({
                    p: 2,
                    height: '100%',
                    width: '100%',
                    '&:hover': {
                      backgroundColor: (theme.vars || theme).palette.action.hover,
                    },
                  }),
                  selectedItemIndex === index && {
                    backgroundColor: 'action.selected',
                  },
                ]}
              >
                <Box
                  sx={[
                    {
                      width: '100%',
                      display: 'flex',
                      flexDirection: 'column',
                      alignItems: 'left',
                      gap: 1,
                      textAlign: 'left',
                      textTransform: 'none',
                      color: 'text.secondary',
                    },
                    selectedItemIndex === index && {
                      color: 'text.primary',
                    },
                  ]}
                >
                  {icon}

                  <Typography variant="h6">{title}</Typography>
                  <Typography variant="body2">{description}</Typography>
                </Box>
              </Box>
            ))}
          </Box>
        </div>
        <Box
          sx={{
            display: { xs: 'none', sm: 'flex' },
            width: { xs: '100%', md: '70%' },
            height: 'var(--items-image-height)',
          }}
        >
          <Card
            variant="outlined"
            sx={{
              height: '100%',
              width: '100%',
              display: { xs: 'none', sm: 'flex' },
              pointerEvents: 'none',
            }}
          >
            <Box
              sx={(theme) => ({
                m: 'auto',
                width: 420,
                height: 500,
                backgroundSize: 'contain',
                backgroundImage: 'var(--items-imageLight)',
                ...theme.applyStyles('dark', {
                  backgroundImage: 'var(--items-imageDark)',
                }),
              })}
              style={
                items[selectedItemIndex]
                  ? {
                      '--items-imageLight': items[selectedItemIndex].imageLight,
                      '--items-imageDark': items[selectedItemIndex].imageDark,
                    }
                  : {}
              }
            />
          </Card>
        </Box>
      </Box>
    </Container>
  );
}
