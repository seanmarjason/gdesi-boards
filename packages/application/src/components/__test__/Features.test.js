import { render } from '@testing-library/react'
import Features from '../Features';

it('Features renders correctly', () => {
  const { container } = render(<Features />)
  expect(container).toMatchSnapshot()
})
