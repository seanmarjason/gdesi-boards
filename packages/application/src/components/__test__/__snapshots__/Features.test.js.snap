// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features renders correctly 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-xkgqcd-MuiContainer-root"
    id="features"
  >
    <div
      class="MuiBox-root css-1cxhys7"
    >
      <h2
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1tydu15-MuiTypography-root"
      >
        Product features
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1fdgtl5-MuiTypography-root"
      >
        Provide a brief overview of the key features of the product. For example, you could list the number of features, their types or benefits, and add-ons.
      </p>
    </div>
    <div
      class="MuiBox-root css-1f9uce6"
    >
      <div>
        <div
          class="MuiBox-root css-ltn6y3"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiBox-root css-1d11eu5-MuiButtonBase-root-MuiButton-root"
            id=":r0:"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root css-1ntlpld"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="ViewQuiltRoundedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 6v4.5c0 .55-.45 1-1 1h-9.67c-.55 0-1-.45-1-1V6c0-.55.45-1 1-1H20c.55 0 1 .45 1 1m-6.33 12v-4.5c0-.55-.45-1-1-1h-3.33c-.55 0-1 .45-1 1V18c0 .55.45 1 1 1h3.33c.55 0 1-.45 1-1m1-4.5V18c0 .55.45 1 1 1H20c.55 0 1-.45 1-1v-4.5c0-.55-.45-1-1-1h-3.33c-.56 0-1 .45-1 1M8.33 18V6c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h3.33c.56 0 1-.45 1-1"
                />
              </svg>
              <h6
                class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
              >
                Dashboard
              </h6>
              <p
                class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
              >
                This item could provide a snapshot of the most important metrics or data points related to the product.
              </p>
            </div>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiBox-root css-1bnwcwe-MuiButtonBase-root-MuiButton-root"
            id=":r1:"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root css-q8sgpl"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="EdgesensorHighRoundedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 7c.55 0 1 .45 1 1v5c0 .55-.45 1-1 1s-1-.45-1-1V8c0-.55.45-1 1-1m-3 3c.55 0 1 .45 1 1v5c0 .55-.45 1-1 1s-1-.45-1-1v-5c0-.55.45-1 1-1m22-3c.55 0 1 .45 1 1v5c0 .55-.45 1-1 1s-1-.45-1-1V8c0-.55.45-1 1-1m-3 3c.55 0 1 .45 1 1v5c0 .55-.45 1-1 1s-1-.45-1-1v-5c0-.55.45-1 1-1m-4-7.99L8 2c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V4c0-1.1-.9-1.99-2-1.99M16 17H8V7h8z"
                />
              </svg>
              <h6
                class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
              >
                Mobile integration
              </h6>
              <p
                class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
              >
                This item could provide information about the mobile app version of the product.
              </p>
            </div>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiBox-root css-1bnwcwe-MuiButtonBase-root-MuiButton-root"
            id=":r2:"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root css-q8sgpl"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="DevicesRoundedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 7c0-.55.45-1 1-1h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-1.1 0-2 .9-2 2v11h-.5c-.83 0-1.5.67-1.5 1.5S.67 20 1.5 20H14v-3H4zm19 1h-6c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h6c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1m-1 9h-4v-7h4z"
                />
              </svg>
              <h6
                class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
              >
                Available on all platforms
              </h6>
              <p
                class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
              >
                This item could let users know the product is available on all platforms, such as web, mobile, and desktop.
              </p>
            </div>
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root css-eld7w0"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-f0nqsc-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiBox-root css-1ujgryp"
            style="--items-imageLight: url("https://mui.com/static/images/templates/templates-images/dash-light.png"); --items-imageDark: url("https://mui.com/static/images/templates/templates-images/dash-dark.png");"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
